flight/Modules/Actuator/actuator.c

```diff
@@ -340,12 +340,23 @@ static void actuatorTask(void* parameters)
 				continue;
 			}
 
-			if((mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) || (mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_SERVO))
+			if((mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) || (mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_SERVO)) {
 				status[ct] = ProcessMixer(ct, curve1, curve2, &mixerSettings, &desired, dT);
-			else
-				status[ct] = -1;
-
 
+				// If a motor type, additional work to be done
+				if(mixers[ct].type == MIXERSETTINGS_MIXER1TYPE_MOTOR) {
+					if (status[ct] > 0) {
+						// Apply curve fitting, mapping the input to the propeller output.
+						status[ct] = actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_A] *
```

Comment:
This is where I discovered the ```NaN``` problem. ;)

If it's okay, I'd like to leave the bidirectional support for later. It will require a different set of curve fitting constants for bidirectional support and I don't know how I would test that efficiently. Currently ```ProcessMixer()``` saturates the motors' lower value at 0, so in theory we should never be able to have a negative value at this point. But I put the check in anyway to ensure that we don't accidentally pass a negative number no matter what happens upstream.

P.S. There's a conversation here about the fact that motors are treated as things that can only run forward, and I think we agree that it's a shame that this assumption is sort of baked into our process.